lifecycle-operator/controllers/lifecycle/keptntask/job_utils.go

```diff
@@ -91,6 +91,7 @@ func (r *KeptnTaskReconciler) generateJob(ctx context.Context, task *klcv1alpha3
 				},
 				Spec: corev1.PodSpec{
 					RestartPolicy: "OnFailure",
+					ServiceAccountName: definition.Spec.ServiceAccount.Name,
```

Comment:
since the ServiceAccount in the Spec is a pointer, please add a check to verify that the pointer is not nil before accessing the property. otherwise, we might encounter crashes due to nil pointer accesses